xmppserver/src/test/java/org/jivesoftware/util/CertificateManagerTest.java

```diff
@@ -299,6 +320,105 @@ public void testServerIdentitiesXmppAddrAndDNS() throws Exception
         assertFalse( serverIdentities.contains( subjectCommonName ) );
     }
 
+    /**
+     * Tests a PEM generated by OpenSSL using this config file:
+     *
+     * <code>
+     * [ req ]
+     * default_bits       = 2048
+     * distinguished_name = req_distinguished_name
+     * req_extensions     = req_ext
+     * x509_extensions    = v3_ca # The main difference
+     * prompt = no
+     *
+     * [ req_distinguished_name ]
+     * C = US
+     * ST = YourState
+     * L = YourCity
+     * O = YourOrganization
+     * OU = YourUnit
+     * CN = yourdomain.com
+     *
+     * [ req_ext ]
+     * subjectAltName = @alt_names
+     *
+     * [ v3_ca ]
+     * subjectAltName = @alt_names
+     * basicConstraints = CA:TRUE
+     * keyUsage = digitalSignature, keyEncipherment
+     * extendedKeyUsage = serverAuth, clientAuth
+     *
+     * [ alt_names ]
+     * otherName.0 = 1.3.6.1.5.5.7.8.7;IA5:_xmpp-server.service.example.com
+     * otherName.1 = 1.3.6.1.5.5.7.8.7;IA5:_dns.service.example.net
```

Comment:
Why is this `assertFalse`d later?
I don't doubt that it should be, it's just not clear from the test.

Reply:
Yeah, these assertions might benefit from some explanatory comments. In this case, it's a valid sRVName for the wrong service.

Reply:
That's a good suggestion. I've now added failure messages to the assertions that describe the (un)desired behavior.